Warnings during compilations

During compilation as is or in project, there are a lot of warnings (if I included `log.h`) with the following messages:

```
In file included from /path/to/logger/log.hpp:5:
In file included from /path/to/logger/logger.hpp:7:
/path/to/logger/appenders/appender_interface.hpp:19:55: warning: field 'appender_type_' will be initialized after field 'severity_' [-Wreorder-ctor]
    IAppender(AppenderType type, Severity severity) : appender_type_(type), severity_(severity) {}
                                                      ^~~~~~~~~~~~~~~~~~~~  ~~~~~~~~~~~~~~~~~~~
                                                      severity_(severity)   appender_type_(type)
/path/to/logger/appenders/appender_interface.hpp:24:39: warning: unused parameter 'severity' [-Wunused-parameter]
    virtual void set_colours(Severity severity, const MessageColours& msg_cols) {}
                                      ^
/path/to/logger/appenders/appender_interface.hpp:24:71: warning: unused parameter 'msg_cols' [-Wunused-parameter]
    virtual void set_colours(Severity severity, const MessageColours& msg_cols) {}
```

Add `-Werror` flag for compilation.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,7 +6,10 @@ set(CMAKE_CXX_STANDARD 17)
 include_directories(include)
 include_directories(include/appenders)
 include_directories(include/formatters)
+
 add_library(liblogger src/main.cpp)
 set_target_properties(liblogger PROPERTIES PREFIX "")
 target_include_directories(liblogger PUBLIC include include/appenders include/formatters)
+
+set(CMAKE_CXX_FLAGS "-Wall -Wextra -Werror")
 add_executable(logger src/main.cpp)

--- a/include/logger/appenders/appender_interface.hpp
+++ b/include/logger/appenders/appender_interface.hpp
@@ -16,12 +16,13 @@ enum AppenderType {
 class IAppender {
 public:
     IAppender(AppenderType type) : appender_type_(type) {}
-    IAppender(AppenderType type, Severity severity) : appender_type_(type), severity_(severity) {}
+    IAppender(AppenderType type, Severity severity) : severity_(severity), appender_type_(type) {}
+    virtual ~IAppender() = default;
     virtual void write(const Record& record) = 0;
     Severity severity() const { return severity_; }
     void set_severity(Severity severity) { severity_ = severity; }
     AppenderType type() const { return appender_type_; }
-    virtual void set_colours(Severity severity, const MessageColours& msg_cols) {}
+    virtual void set_colours(Severity /*unused*/, const MessageColours& /*unused*/) {}
     virtual void turn_colours_on() {}
     virtual void turn_colours_off() {}
 //    virtual ~IAppender(); // TODO: See Vladimirov's lectures

